### PR TITLE
OPS-1327 Allow User to specify the lifetime of the sandbox when creating

### PR DESCRIPTION
**Pull request by arbabnazar**
@edx/devops kindly review. Thanks

Reference:

http://xxxxx.edx.org:8080/job/ansible-provision-2/2/console

### CODE REVIEW

File: util/jenkins/ansible-provision.sh [hunk 1]
```diff
@@ -37,7 +37,7 @@ run_ansible() {
 }
 
 # This DATE_TIME will be used as instance launch time tag
-TERMINATION_DATE_TIME=`date +"%m-%d-%Y %T" --date="-7 days ago"`
+TERMINATION_DATE_TIME=`date +"%m-%d-%Y %T" --date="-${sandbox_life} days ago"`
```
Comment (feanil) [inline util/jenkins/ansible-provision.sh]: We should set a default for this if it is not set.  `if [[ -z $sandbox_life ]]`

Comment (maxrothman) [inline util/jenkins/ansible-provision.sh]: Actually, that will also fail if `sandbox_life` isn't defined. You could quote it, e.g. `[[ -z "$sandbox_life" ]]`, but I think the better option would be to use `${sandbox_life-default}` ([ref](http://www.tldp.org/LDP/abs/html/parameter-substitution.html#PARAMSUBREF)). If `sandbox_life` is used elsewhere in the script, you might instead want to use `${sandbox_life=default}`.

**Comment by arbabnazar** [comment]
We have default value of it( 8 days) from jenkins. Thanks

**Comment by arbabnazar** [comment]
@e0d kindly review it, I have added the validation check for the number of days. Thanks

**Comment by arbabnazar** [comment]
@feanil can you please review this, the default value of this variable will come from here:
screen shot 2016-02-17 at 5 58 14 pm
Thanks,

File: util/jenkins/ansible-provision.sh [hunk 2]
```diff
@@ -37,7 +37,13 @@ run_ansible() {
 }
 
 # This DATE_TIME will be used as instance launch time tag
-TERMINATION_DATE_TIME=`date +"%m-%d-%Y %T" --date="-7 days ago"`
+if [[ ! -n ${sandbox_life//[0-9]/} ]]  && [[ ${sandbox_life} -le 30 ]]; then
+    TERMINATION_DATE_TIME=`date +"%m-%d-%Y %T" --date="-${sandbox_life} days ago"`
```
Comment (fredsmith) [inline util/jenkins/ansible-provision.sh]: this wording is confusing to read - why not just say "${sandbox_life} days"?

```
$ date +"%m-%d-%Y %T" --date "8 days"
02-25-2016 08:57:55
```

**Comment by arbabnazar** [comment]
@fredsmith fix the message, kindly review. Thanks

**Comment by maxrothman** [comment]
As per @feanil's previous comment, there should be a default if `sandbox_life` is not set.

**Comment by arbabnazar** [comment]
@edx/devops kindly review. Thanks

**Comment by maxrothman** [comment]
@arbabnazar as per my previous comment, you can get rid of the if-statement if you use `${sandbox_life-default}` or `${sandbox_life=default}`.

**Comment by arbabnazar** [comment]
fixed, thanks for the tips @maxrothman

**Comment by maxrothman** [comment]
:+1:

File: util/jenkins/ansible-provision.sh [hunk 3]
```diff
@@ -37,7 +37,13 @@ run_ansible() {
 }
 
 # This DATE_TIME will be used as instance launch time tag
-TERMINATION_DATE_TIME=`date +"%m-%d-%Y %T" --date="-7 days ago"`
+if [[ ! -n ${sandbox_life//[0-9]/} ]]  && [[ ${sandbox_life} -le 30 ]]; then
+    TERMINATION_DATE_TIME=`date +"%m-%d-%Y %T" --date="-${sandbox_life:-7} days"`
```
Comment (fredsmith) [inline util/jenkins/ansible-provision.sh]: you took out the 'ago', but you left the negative.  This won't do what you want.

Also, your default has a negative in it, and you're prefixing your string with a negative, so make sure you get rid of both.

**Comment by arbabnazar** [comment]
@fredsmith thanks, I have fixed it as per your suggestion.

**Pull request merged**